### PR TITLE
Scrape team information from Notion 

### PR DESCRIPTION
**Pull request by dfarooq610**
Adds a small script to fetch information from Notion's club directory. The access token can be found in the EBoard Bitwarden. 

you also need a database id as an environment variable. the URL for the notion club directory may look like this https://www.notion.so/sandboxnu/abc?v=def -- your database id would be `abc`. 

this will update the team.json file in src/content/team to have a bunch of stubs like this 
```json
{
      "name": "Luke Taylor",
      "team": {
        "name": "GraduateNU",
        "role": "Developer"
      },
      "profileImage": "./profileImages/TaylorLuk.jpg",
      "socialMedia": {
        "email": "taylor.luk@northeastern.edu",
        "linkedIn": "https://www.linkedin.com/in/luke-felix-taylor/",
        "portfolio": "https://lukefelixtaylor.com"
      }
}
```

so we run this, make a PR, and then everyone uploads their headshot into the ./src/content/teams/profileImages folder -- the file is named after their email -- if their email is last.first@northeastern.edu, the name of the file is `LastFirst.jpg`. 

role to position mapping is a bit naive, so would probably just need validation. 

so updating in the future is as simple as running `yarn run team-scrape`.
image

### Current limitations
the data structure for this page is not great, like there are multiple Allys and Shashs on the page since y'all are on multiple proj teams. There is probably a way to optimize the amount of duplicate info in the JSON file, but that's a problem for later

this was just a small tool to make the current system a bit easier to navigate -- it's unfortunately not optimizing the team page workflow

### CODE REVIEW

**Comment by dfarooq610** [comment]
waiting for Da-Jin to integrate the sandbox integration into the club directory table.

File: src/utils/teamScraper.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+                    "does_not_contain": "Hiatus"
+                }
+            }
+        ]
+    },
+}
+eboard = set(["Executive Director", "E-Board Liason", "Technical Director",
+             "UX Director", "Marketing Director", "Operations Director"])
+head_ofs = set(["Head of Recruiting", "Head of UX",
+               "Head of Project Acquisition"])
```
Comment (dillydally414) [inline src/utils/teamScraper.py:37]:
```suggestion
head_ofs = set(["Head of Recruiting", "Head of UX",
               "Head of Project Acquisition", "Head-of Community", "Head of DX"])
```

File: src/utils/teamScraper.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    social_media_filtered = {k: v for k,
+                             v in social_media.items() if v is not None}
+
+    return {
+        "name": name,
+        "team": {
+            "name": team_name,
+            "role": role,
+        },
+        # "profileImage": "" if email is None else './profileImages/' + ''.join([name.capitalize() for name in email[0: email.index('@')].split('.')]) + ".jpg",
```
Comment (dfarooq610) [inline src/utils/teamScraper.py:54]: If we can use Pillow to get profile image from notion, script can then save it to a file locally, naming convention in the PR summary above

**Comment by shashjar** [comment]
@dillydally414 please review

File: src/utils/teamScraper.py [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+                      "Operations Director": 3, "Marketing & Events Director": 4, "E-Board Liaison": 5}
+head_of_roles = set(["Head of Recruiting", "Head of DX", "Head of UX",
+                     "Head of Project Acquisition", "Head of Community"])
+head_of_roles_order = {"Head of Recruiting": 0, "Head of DX": 1, "Head of UX": 2,
+                       "Head of Project Acquisition": 3, "Head of Community": 4}
+brand_roles_order = {"Marketing & Events Director": 0, "Brand Designer": 1}
+project_teams = set(["Carpool", "Faculty Activity Tracker", "GraduateNU",
+                    "HappyEastie", "MFA", "ScoutTrek", "SearchNEU", "SGA Tooling"])
+project_team_member_order = {"Project Lead": 0,
+                             "Design Lead": 1, "Designer": 2, "Developer": 3}
```
Comment (dillydally414) [inline src/utils/teamScraper.py:39]:
```suggestion
project_team_member_order = {"Project Lead": 0,
                             "Design Lead": 1, "Technical Lead": 2, "Designer": 3, "Developer": 4}
```

File: src/utils/teamScraper.py [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+        "team": {
+            "name": team_name,
+            "role": role,
+        },
+        "profileImage": f"./profileImages/SP23/{name}.png",
+        "socialMedia": social_media_filtered
+    }
+
+
+def generate_team_mappings(roles, teams):
```
Comment (dillydally414) [inline src/utils/teamScraper.py:73]: Somewhere in here we need to account for Tech Leads. Not sure whether we should just remove Developer role from them in the Notion or if you want to deal with parsing the multiple role case here, but right now both SGA and ScoutTrek are missing the acknowledgement of Chan Mi and Ross respectively.

**Review by dillydally414 — CHANGES_REQUESTED** [review]
Looks good except for tech leads, some brief comments might be nice in the scraper to make it easier to modify or extend in the future.

File: src/utils/teamScraper.py (hunk 4 above)
Comment (shashjar) [reply src/utils/teamScraper.py:73]: removed Developer role from both Ross and Chan Mi on Notion and re-updated the team JSON file

**Review by dillydally414 — APPROVED** [review]
LGTM!

**Pull request merged**